### DISCUSSION TITLE
Guidance for tuning combos for false positives/negatives?

### DISCUSSION THREAD

**Discussion by rcurran1221** [urob/zmk-config#79] [Q&A] [answered]
i am currently set up to use a require-prior-idle-ms = 150 ms, with timeout-ms = 18 (same config for horizontal combos that this repo uses), and i am wondering how would i tune these settings to decrease false positives (combo firing when expecting key taps)?

i believe i would want to decrease timeout-ms if i encounter false positives, but wanted to confirm! thank you!

**Comment by urob** [comment, marked as answer]
Yes, reducing the combo term and increasing the require-prior-idle term should help. 

The former defines the interval within which both keys have to be pressed to trigger the combo. There is no downside of reducing it as long as you can reliably trigger the combo when intended. 

The latter defines the idle time you have to wait after any other key press for a combo to trigger. Increasing this value may also prevent combos going off accidentally in the middle of typing bursts. The downside is that you will need to pause for that amount of time when intentionally trying to trigger combos.

**Comment by rcurran1221** [comment]
thank you!

do you know of any diagnostics/logging i could run to see if i fire a false negative because of the "timeout-ms" setting or the "require-prior-idle-ms" settings?

**Reply by urob** [reply]
Hm, with the kind of behaviors I bind to combos it's usually apparent from just what happens on the screen. But if you want to more systematically log the output any kind of keylogger will do (I usually use something home-baked in AHK). Or if you want to rigorously look into what's going on on your board, there's also usb logging for zmk boards.
